news/tests/test_api.py

```diff
@@ -1,59 +1,142 @@
 from rest_framework.test import APITestCase
 
-from utils.setup_tests.factory_class import (
-    AuthorFactory,
-    CategoriesFactory,
-    CommentFactory,
-    NewsApprovedFactory,
-    NewsFactory,
-    NewsUserFactory,
-    TagsFactory,
-)
+from utils.setup_tests.author import AuthorFactory
+from utils.setup_tests.category import CategoriesFactory
+from utils.setup_tests.comment import CommentFactory
+from utils.setup_tests.news import ApprovedNewsFactory, AuthoredNewsFactory, NewsFactory
+from utils.setup_tests.news_user import NewsUserFactory
+from utils.setup_tests.saved_news import SavedNewsFactory
+from utils.setup_tests.tags import TagsFactory
 
 
 class TestCategories(APITestCase):
     @classmethod
     def setUpTestData(cls):
         cls.categories = CategoriesFactory()
+        cls.news = NewsFactory()
 
     def test_get_list_categories(self):
         response = self.client.get("/api/Categories/", format="json")
         self.assertEqual(response.status_code, 200)
         self.assertEqual(self.categories.title, response.json()[0]["title"])
 
+    def test_success_get_news_by_categories(self):
+        title = self.news.title.split()[0]
+        response = self.client.get(f"/api/Categories/{title}/", format="json")
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(self.news.title, response.json()[0]["title"])
+        self.assertEqual(self.news.description, response.json()[0]["description"])
+        self.assertEqual(self.news.content, response.json()[0]["content"])
+
+    def test_get_news_by_categories_not_found(self):
+        title = "rrrrrrrrrrrrrrrrrrr"
```

Comment:
what a hell is that?You must do self.categories.title or self.categories[0].title